sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -637,6 +638,17 @@ object DataSource extends Logging {
                     "Hive built-in ORC data source must be used with Hive support enabled. " +
                     "Please use the native ORC data source by setting 'spark.sql.orc.impl' to " +
                     "'native'")
+                } else if (provider1.toLowerCase(Locale.ROOT) == "avro" ||
+                  provider1 == "com.databricks.spark.avro" ||
+                  provider1 == "org.apache.spark.sql.avro") {
+                  throw new AnalysisException(
+                    s"Failed to find data source: $provider1. Avro is built-in data source since " +
+                    "Spark 2.4. Please deploy the application as per " +
+                    s"$latestDocsURL/avro-data-source-guide.html#deploying")
```

Comment:
Please merge https://github.com/apache/spark/pull/22121 before this one is merged.
But first we need to have agreement on the configuration name, since it is also mentioned in the documentation.

Reply:
We can update the message later. No need to be blocked by that.

Reply:
perhaps we should say built-in but external module.  As if its built-in I would expect it to automatically be there.

Reply:
@tgravescs Make sense, thank you.